docusaurus/docs/iOS/client/state-layer/state-layer-overview.md

```diff
@@ -0,0 +1,233 @@
+---
+title: State Layer
+---
+
+:::note
+Since `StreamChat` version 4.55.0.
+:::
+
+The `StreamChat` framework represents the Stream API with [controllers](../controllers/controllers-overview.md) which use completion handlers and delegates for communicating about state changes. In addition to controllers, we provide a new and modern way of interacting with the Stream API. It follows an architecture where we have objects interacting with the Stream API through async functions. These objects are accompanied with state objects that hold the current state. This architecture follows similar patterns as our [Video SDK for iOS](https://getstream.io/video/sdk/ios/).
```

Comment:
It could be interesting to add, that his new state layer is the future, and on v5, it will be the only way to interact with the SDK, and the controller will be deprecated 🤔

Reply:
We don't know that 100% to write it publicly, it depends on the context when v5 is released and would probably need an approval from the business side.

Reply:
I think we should at least say that the State layer should be the preferrable way, and that we recommend using first the state layer